yaksh/views.py

```diff
@@ -1698,3 +1711,95 @@ def design_questionpaper(request):
         context = {'form': form}
         return my_render_to_response('yaksh/design_questionpaper.html',
                                      context, context_instance=ci)
+
+@login_required
+def view_profile(request):
+    """ view moderators and users profile """
+
+    context = {}
+    user = request.user
+    ci = RequestContext(request)
+    if not user.is_authenticated():
+        raise Http404('You are not allowed to view this page!')
+    else:
+        profile_details = Profile.objects.get(user_id=user.id)
+        user_details = User.objects.get(id=user.id)
```

Comment:
you already have user object with you, so no need to get it.
Also, user.profile can give profile or vice versa, so need to fetch the objects.